modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -588,6 +597,8 @@ public ObjectField updateObjectField(
 	public ObjectField updateRequired(long objectFieldId, boolean required)
 		throws PortalException {
 
+		_validateRelationship(objectFieldId, required);
```

Comment:
Can you move this validation to after getting the objectField than only call it if the field has a relationshipType?